### DISCUSSION TITLE
Why is SwiftDocCPlugin included by default?

### DISCUSSION THREAD

**Discussion by guitaripod** [onevcat/Kingfisher#1952] [Q&A] [answered]
What is the benefit for me who wants to use Kingfisher only?
<img width="182" alt="CleanShot 2022-06-13 at 21 05 09@2x" src="https://user-images.githubusercontent.com/47460844/173416557-ffd5296f-7d18-4019-9e85-19d27836441b.png">

**Comment by onevcat** [comment, marked as answer]
Ummm, this is for generating the DocC documentation [here](https://swiftpackageindex.com/onevcat/Kingfisher/master/documentation/kingfisher). 

Indeed it is not a necessary for framework users, but it has to be existing as a dependency of the package to allow Swift Index to build the doc and continue the following process. Unfortunately, Xcode will just also fetch it and show in the dependencies tree. But the good news is this plugin package is not added to your runtime at all and there is a system-wide cache so you only download it once.

@daveverwer, Not sure if it is a good way to mention you here, but do you have any comment on this? Is there a recommended way for users to not containing the DocC plugin in their project? I am now following your instruction to add the plugin like this:

```swift
#if swift(>=5.6)
  // Add the documentation compiler plugin if possible
  package.dependencies.append(
    .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0")
  )
#endif
```

Maybe there is a chance to have some additional compilation flag or environment values to refine the dependency better for framework user?

**Reply by guitaripod** [reply]
Thanks for looking into it ❤️

**Reply by daveverwer** [reply]
We published an update to the blog post last week with some more information on this. You can find it here: https://blog.swiftpackageindex.com/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation/

The good news is that it's no longer necessary to include that plugin if you want SPI to generate your docs:

> UPDATE: It’s no longer necessary to add the DocC plugin to your package. If you’ve added it already, feel free to remove it. Our build system will also cope if you leave the plugin or need it for another purpose, so there’s no rush to remove it! The only change you need to make to your project is to [add the manifest file](https://blog.swiftpackageindex.com/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation/#add-a-spi-manifest).

Hope this helps!

**Reply by guitaripod** [reply]
Thanks, Dave

**Reply by onevcat** [reply]
@daveverwer Great! I will soon remove it and schedule a new tag for it. Please keep an eye!

**Reply by onevcat** [reply]
PR done: #1954 and will be release as 7.2.4. Another resolving once the new version comes out would remove DocC plugin from your dependency tree.
